src/chainrules.jl

```diff
@@ -0,0 +1,94 @@
+## Uniform ##
+
+function ChainRulesCore.rrule(::typeof(uniformlogpdf), a, b, x)
+    diff = b - a
+    insupport = a <= x <= b
+    lp = insupport ? -log(diff) : log(zero(diff))
+
+    function uniform_logpdf_pullback(ȳ)
+        z = zero(x) * ȳ
+        if a <= x <= b
+            c = ȳ / diff^2
```

Comment:
Shouldn't it be this?
```suggestion
            c = ȳ / diff
```

Reply:
Good catch! I just copied the existing code without thinking about the mathematical details at all. I guess it went undetected since unfortunately the tests for univariate distributions still use "nice" parameters and samples (for multivariate + matrixvariate I changed this a month ago which revealed some bugs as well) and in particular the uniform distribution was only tested on the unit interval.